push the current branch before retargetting

isaacs opened the pull request:
A few times we've had 'pull update' fail because the remote branch
we want to target doesn't yet exist.  This pushes the curbranch
up to origin before trying to retarget.

<!-- What / Why -->
<!-- Describe the request in detail. What it does and why it's being changed. -->


## References
<!-- Examples:
  Related to #0
  Depends on #0
  Blocked by #0
  Fixes #0
  Closes #0
-->

isaacs committed:
push the current branch before retargetting

A few times we've had 'pull update' fail because the remote branch
we want to target doesn't yet exist.  This pushes the curbranch
up to origin before trying to retarget.

ruyadorno commented on pull.sh:
```diff
@@ -193,6 +193,10 @@ update () {
 
+  # do an initial push just to make sure it exists, otherwise
+  # the retargetting will fail.
+  git push origin "$curbranch"
```
Actually the bit that pushes to the current branch has extra logic to check if we're currently sitting in a default branch in order to avoid pushing (just wanted to be extra safe in that case, ref: https://github.com/npm/pull/pull/5/files#diff-df027a58eb34e6f4541846cb4f7be10824a342ab71b4f006a18de21a14f90a76R215)

So for the sake of sticking to that idea I think it's better to also wrap this `git push` in a similar way:

```suggestion
  if [ "$curbranch" != "" ] || \
     [ "$curbranch" != "$defaultbranch" ] || \
     [ "$curbranch" != "master" ] || \
     [ "$curbranch" != "main" ] || \
     [ "$curbranch" != "latest" ]; then
    git push origin "$curbranch"
  fi
```

ruyadorno replied:
either that OR just being fine with it pushing to master/latest/main and removing the conditional logic from the later `git push`

isaacs replied:
Ahh, ok.  Good call.

Maybe we need a `should_push "$branch"` function, so we could just do:

```bash
if should_push "$curbranch"; then
  git push origin $curbranch
fi
```

isaacs committed:
output style updates

Output for humans should go to &2, not &1
Emojis are cute, but they're breaking my terminal sometimes 😬

isaacs commented:
Ok, added a `should_push` method.  Also added a second commit to send human output (usage etc) to stderr, and remove emojis because iTerm2 seems to be making them really weird with my current setup for some reason 😬

isaacs committed:
Get the reviewers from github API

PR-URL: https://github.com/npm/pull/pull/5
Credit: @isaacs
Close: #5
Reviewed-by: @ruyadorno
